nomad/server.go

```diff
@@ -96,20 +99,29 @@ type Server struct {
 	leaderCh      <-chan bool
 	raft          *raft.Raft
 	raftLayer     *RaftLayer
-	raftStore     *raftboltdb.BoltStore
-	raftInmem     *raft.InmemStore
-	raftTransport *raft.NetworkTransport
+	raftLayerLock sync.Mutex
```

Comment:
Why do we need all these locks? The agent can only process one reload at a time and the reload is the only thing that modifies these after creation